### ISSUE TITLE
Programs that use ncurses do not resize dynamically

### ISSUE THREAD

**Issue by uni-dos** [getsolus/packages#2673]
### Please confirm there isn't an existing open bug report

- [X] I have searched open bugs for this issue

### Summary

Applications that use ncurses such as htop, nvtop, and radeontop have their contents weirdly rendered when the terminal is resized.

### Steps to reproduce

1. Open htop/nvtop.
2. Resize the window

### Expected result

The TUI should resize to the appropriate size of the terminal. 

### Actual result

The TUI contents do not take the full space of the terminal when maximized. When TUI apps are opened in a maximized terminal and resized the contents are not properly arranged.

### Environment

- [X] Is system up to date?

### Repo

Shannon (stable)

### Desktop Environment

Plasma

### System details

System:
  Host: ryzentop Kernel: 6.8.10-291.current arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.0.4 Distro: Solus 4.5 resilience
Machine:
  Type: Desktop System: Micro-Star product: MS-7C92 v: 1.0
    serial: <superuser required>
  Mobo: Micro-Star model: MPG B550I GAMING EDGE MAX WIFI (MS-7C92) v: 1.0
    serial: <superuser required> UEFI: American Megatrends LLC. v: 1.F0
    date: 11/01/2023
CPU:
  Info: 8-core AMD Ryzen 7 5800X3D [MT MCP] speed (MHz): avg: 3517
    min/max: 550/4550
Graphics:
  Device-1: AMD Navi 23 [Radeon RX 6600/6600 XT/6600M] driver: amdgpu
    v: kernel
  Display: wayland server: X.org v: 1.21.1.13 with: Xwayland v: 23.2.6
    compositor: kwin_wayland driver: N/A resolution: 1920x1080
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.0.7 renderer: AMD
    Radeon RX 6600 XT (radeonsi navi23 LLVM 16.0.6 DRM 3.57
    6.8.10-291.current)
Network:
  Device-1: Intel Wi-Fi 6 AX200 driver: iwlwifi
  Device-2: Realtek RTL8125 2.5GbE driver: r8169
Drives:
  Local Storage: total: 2.73 TiB used: 101.73 GiB (3.6%)
Info:
  Memory: total: 32 GiB available: 31.28 GiB used: 3.83 GiB (12.2%)
  Processes: 374 Uptime: 26m Shell: Bash inxi: 3.3.34

**Comment by ReillyBrogan** [comment]
I can't reproduce this with konsole on Plasma Wayland. Is this still reproducible for you after the latest Qt and Plasma updates?

**Comment by uni-dos** [comment]
> I can't reproduce this with konsole on Plasma Wayland. Is this still reproducible for you after the latest Qt and Plasma updates?

Yes, it's still reproducible after updating.

**Comment by ReillyBrogan** [comment]
I need a bit more reproduction details then. Can you take a screencast of it while it's happening?

**Comment by uni-dos** [comment]
[Screencast_20240610_144027.webm](https://github.com/getsolus/packages/assets/32173659/da0d30e9-6e73-4b84-8859-7f92d75b7638)

**Comment by andreibuh41** [comment]
I am having the same bug on budgie.
Tested with solus 4.5 iso fresh install and the bug is not present.
After updating to the latest packages the bug is present.

To reproduce: open nano and try to resize the terminal

**Comment by Staudey** [comment]
I can only reproduce when I switch to `bash` (my usual shell is `zsh`)

**Comment by TraceyC77** [comment]
I am not able to reproduce this with zsh or fish. I can reproduce it regardless of terminal application, with bash
In bash, the problem can be reproduced with any cli application

**Comment by andreibuh41** [comment]
I installed zsh and run zsh and I can still reproduce the bug.
I also tried a couple of terminal from flathub and still can reproduce the bug.

**Comment by TraceyC77** [comment]
The last bash related pull request was getsolus/packages/pull/2670

**Comment by uni-dos** [comment]
If you launch htop or nvtop from it's desktop entry, they are able to dynamically resize.

**Comment by uni-dos** [comment]
Yes, it is still present.

**Comment by uni-dos** [comment]
So I tested this again with `host-spawn` and htop and nvtop were able to reszie. So I thought the issue is related to an environment variable. I did use AI to help be determine it and it recommended me I unset `LINES` and `COLUMNS`. After unsetting these for the current bash instance I was able to resize htop without it staying the same or looking messed up

**Comment by TraceyC77** [comment]
Thanks for following up, and glad to hear you got it solved.

**Comment by uni-dos** [comment]
Can this be reopened. I tested this with the Solus 4.9 ISO and I am still able to reproduce this issue.

**Comment by TraceyC77** [comment]
> Can this be reopened. I tested this with the Solus 4.9 ISO and I am still able to reproduce this issue.

I'm still not able to reproduce this with Konsole running htop or nvtop.
Are you able to reproduce this with a brand new user with no custom settings?

**Comment by uni-dos** [comment]
Yes, I was using just the live installer in virtmanager. This also happens with AerynOs.

**Comment by TraceyC77** [comment]
Well, darn. I'll reopen this. Since it happens with Solus and AerynOS, there must be something in common in the packaging somewhere.

ping @ReillyBrogan

**Comment by TraceyC77** [comment]
I can confirm that while I can't reproduce with `zsh` in Konsole, I can with `bash`, running `htop`. The program does not resize with the Konsole window.

Terminology and Wezterm are also affected, this looks to be unaffected by the terminal program used

**Comment by uni-dos** [comment]
Yeah it happens regardless of what terminal I used. I thought this was a possible wayland issue but it is also present using xfce.

**Comment by uni-dos** [comment]
I don't have zsh set up as my default shell and I am able to reproduce the issue.

**Comment by ReillyBrogan** [comment]
I messed around with this for a while and as far as I can tell this is bash-specific and a result of bash enabling `checkwinsize` by default with Bash 5.

From the docs:
> If set, Bash checks the window size after each external (non-builtin) command and, if necessary, updates the values of LINES and COLUMNS, using the file descriptor associated with stderr if it is a terminal. This option is enabled by default. 

As a result of this the `LINES` and `COLUMNS` envvars are set when you launch htop and when those envvars are present ncurses will ignore the SIGWINCH signal (which is sent by the terminal upon window resizing) and will just continue to use whatever `LINES` and `COLUMNS` are set to.

And indeed if you do `unset LINES COLUMNS && htop` from bash you'll see that resizing works fine, which indicates to me that this is *not* a problem with any of the Solus/Aeryn packages.

Since this is an upstream issue I suggest to close this issue, and open one with either the bash or ncurses devs (or both) instead.

**Comment by uni-dos** [comment]
But that doesn't explain why it works on Fedora and other distros. I tested Fedora right now and htop is able to resize. checkwinsize is enabled as well.

**Comment by TraceyC77** [comment]
`htop` also resizes properly with `bash` on Neon Testing. There's clearly a difference in how we're packaging bash or something.

**Comment by uni-dos** [comment]
As I was updating ncurses I noticed that there is a symbolic link in the package `/usr/lib/terminfo`. It looks like it points to `../share/terminfo` which I believe does not exist. I think this is not an issue with bash but ncurses

**Comment by EbonJaeger** [comment]
`/usr/share/terminfo` is a directory on my system with a lot of stuff in it, owned by `ncurses`.

**Comment by uni-dos** [comment]
I think ncurses has a broken symlink then. `/usr/lib/terminfo` has to point to `/usr/share/terminfo` not `../share/terminfo` I could be wrong

**Comment by EbonJaeger** [comment]
Going to `/usr/lib` and doing `ls -l | grep terminfo` shows `lrwxrwxrwx      - root root  10 Apr 19:13  terminfo -> ../share/terminfo`. Doing `ls -l ../share/terminfo` lists the directory correctly. It seems correct to me.

**Comment by uni-dos** [comment]
Darn I thought I was onto something
